retrieval_service/datastore/providers/firestore.py

```diff
@@ -142,12 +283,49 @@ async def delete_collections(collection_list: list[AsyncCollectionReference]):
                 .set(
                     {
                         "content": policy.content,
-                        "embedding": policy.embedding,
+                        # Vector type does not accept None value
+                        "embedding": Vector(policy.embedding or []),
                     }
                 )
             )
         await asyncio.gather(*create_policies_tasks)
 
+        # Initialize single-field vector indexes
+        create_amenities_vector_index = [
```

Comment:
nit: we could refactor this so we don't have 2 similar lists (1 for amenities and policies). It might be nice to have these outside of this code block